network/src/primary.rs

```diff
@@ -72,17 +75,20 @@ impl PrimaryNetwork {
         message: BincodeEncodedPayload,
     ) -> CancelHandler<()> {
         let client = self.client(address);
-        let handle = tokio::spawn(
-            self.retry_config
-                .retry(move || {
-                    let mut client = client.clone();
-                    let message = message.clone();
-                    async move { client.send_message(message).await.map_err(Into::into) }
-                })
-                .map(|response| {
-                    response.expect("we retry forever so this shouldn't fail");
-                }),
-        );
+        let handle = self
+            .executor
+            .spawn(
+                self.retry_config
+                    .retry(move || {
+                        let mut client = client.clone();
+                        let message = message.clone();
+                        async move { client.send_message(message).await.map_err(Into::into) }
+                    })
+                    .map(|response| {
+                        response.expect("we retry forever so this shouldn't fail");
+                    }),
```

Comment:
Taking notes for a further PR (and @asonnino | @bmwill's eyes): now we can start splitting this out between both callers that land here: `broadcast` (reliable sends we must make sure we hear back from) and `unreliable_(send|broadcast)` (which we don't).

The retry is really here to make sure we execute the future >= 1 in the case of the unreliable send, so we could envision a 2-lane system: the `broadcast`'s `send_message` latch on the semaphore directly as they do now, because their not getitng through would block the protocol, whereas the `unreliable...` would use `try_spawn` and `try_acquire_owned` in a general (mutualized) retry, so as to only queue when there are permits left on the semaphore.